File: Funs/uvi.m
function uvi = uvi(labels, S)

MatLabel = labels - labels'; 

MatLabel = MatLabel == 0; 

MatLabel = double(MatLabel);

uvi = (sum(sum(S.*MatLabel))/sum(MatLabel(:)))/(sum(sum(S.*~MatLabel))/sum(~MatLabel(:))+eps);

end

File: Python/uvi.py
import numpy as np

def uvi(labels, S):
    # Calculate the label matrix
    MatLabel = np.equal.outer(labels, labels)

    # Ensure the matrix is double (float in Python)
    MatLabel = MatLabel.astype(float)

    # Calculate uvi according to the formula
    uvi_value = (np.sum(S * MatLabel) / np.sum(MatLabel)) / (np.sum(S * ~MatLabel) / np.sum(~MatLabel) + np.finfo(float).eps)

    return uvi_value